### PR TITLE
Feat/edit gmc2 johnny

### PR DESCRIPTION
**Pull request by ReubenRobertsHEE**
TIS21-6292

### CODE REVIEW

**Comment by ReubenRobertsHEE** [comment]
@john12321 - is there anything else you would like covered test-wise? I'm guessing you can't review this (seeing as how much you've put into it), but any parting remarks or other concerns?

**Comment by john12321** [comment]
> @john12321 - is there anything else you would like covered test-wise? I'm guessing you can't review this (seeing as how much you've put into it), but any parting remarks or other concerns?

@ReubenRobertsHEE Just seen your e2e test commit. You are the Cypress Sensai 🥋. Was thinking about adding a gmcStatus loading state test to the Profile.cy.tsx but you've covered it much more realistically in the e2e test.

**Comment by Judge40** [comment]
Holy commit history Batman! 😂

File: cypress.config.ts [hunk 1]
```diff
@@ -24,8 +24,8 @@ export default defineConfig({
     supportFile: "cypress/support/component.ts"
   },
   e2e: {
-    baseUrl: "https://stage.trainee.tis.nhs.uk/",
-    // baseUrl: "http://local.tis-selfservice.com",
+    //baseUrl: "https://stage.trainee.tis.nhs.uk/",
+    baseUrl: "http://local.tis-selfservice.com",
```
Comment (Judge40) [inline cypress.config.ts:28]: I'm assuming this needs reverting

File: cypress/e2e/profile/Profile.spec.ts
Comment (Judge40) [inline cypress/e2e/profile/Profile.spec.ts:1]: Is there any value in an additional test which explicitly verifies that an invalid value like `UNKNOWN` is retained if the user cancels the edit request, just to ensure we don't have any weird interactions with the validation?

**Review by Judge40 — CHANGES_REQUESTED** [review]

File: cypress/e2e/profile/Profile.spec.ts
Comment (ReubenRobertsHEE) [reply cypress/e2e/profile/Profile.spec.ts:1]: Sure, I'll push one (and revert that e2e config change that crept in)

**Review by Judge40 — APPROVED** [review]

**Pull request merged**